src/ember-app.js

```diff
@@ -355,6 +383,12 @@ class EmberApp {
       result._destroy();
 
       clearTimeout(destroyAppInstanceTimer);
+
+      if (buildSandboxPerVisit) {
+        // if sandbox was built for this visit, then build a new sandbox for the next incoming request
+        // which is invoked using buildSandboxPerVisit
+        this._pendingNewApplicationInstance = this.buildNewApplicationInstance();
```

Comment:
I might be reading this wrong, but if there are multiple concurrent `visit()` calls then it's possible to call this twice. I think you might need to make sure that `this._pendingNewApplicationInstance` is empty before doing this. If it is already set, then you probably shouldn't do anything.

This leads me to believe that there will be at most one application instance waiting to be used, so if multiple visit calls come at the same time, the second one will need to build the app instance anyways.

Reply:
Thanks for catching that. I'll fix this in a subsequent PR.